Packages/TensorFlow-Examples.pck.st

```diff
@@ -1090,30 +1090,30 @@ initializeInferenceGraph
 		netInput softmax findMaxOn: axis]
 ! !
 
-!MNISTSoftMaxExamplePlan methodsFor: 'initialization' stamp: 'JB 2/11/2017 12:20:32'!
+!MNISTSoftMaxExamplePlan methodsFor: 'initialization' stamp: 'gera 2/12/2017 23:11:51'!
 initializeLearningGraph
-	| learningRate learningRateValue activation gradient |
-	learningRateValue _ 0.9 asTensor.
-	learningRate _ graph const: 'learningRate' value: learningRateValue.
-	activation _ graph
-		fromBlock: [:expected |
-			expectedLabel _ expected.
-			netInput sparseSoftmaxCrossEntropyWithLogits: expected]
-		inputTypes: {TF_Tensor typeInt32}.	
-	activation useOutput: 1.
-	gradient _ activation meanOn: #(0) asInt32Tensor.
-	"todo: divide by the number of inputs (the batch size)"
-	learnWeights _ weights descent: input \* activation rate: learningRate.
-	learnBiases _ biases descent: gradient rate: learningRate
-	! !
-
-!MNISTSoftMaxExamplePlan methodsFor: 'initialization' stamp: 'JB 2/11/2017 14:45:59'!
+	| learningRate learningRateValue gradient |
+	graph
+		fromBlock: [
+			learningRateValue _ 0.9 asTensor.
+			learningRate _ graph const: 'learningRate' value: learningRateValue.
+			activation useOutput: 1.
+			gradient _ activation meanOn: #(0) asInt32Tensor.
+			"todo: divide by the number of inputs (the batch size)"
+			learnWeights _ weights descent: input \* activation rate: learningRate.
+			learnBiases _ biases descent: gradient rate: learningRate]
+		named: 'learning'! !
+
+!MNISTSoftMaxExamplePlan methodsFor: 'initialization' stamp: 'gera 2/12/2017 23:09:05'!
 initializeLossGraph
 	loss _ graph
 		fromBlock: [:expected |
-			expectedForComparison _ expected.
-			(netInput sparseSoftmaxCrossEntropyWithLogits: expected) meanOn: #(0) asInt32Tensor]
-		inputTypes: {TF_Tensor typeInt32}! !
+			expectedLabel _ expected.
```

Comment:
Javi, check how I unified part of the loss calculation path and the learning path.
Both now use the input (expectedLabel) and same activation node, the loss side uses output #0 from the softmax and the learning side uses output #1.

I believe it's still correct, but a bit simpler and a bit more aligned with the ideas of the softmax node, which has two outputs.

If you confirm it's good and like it, merge it :-)

:+1: